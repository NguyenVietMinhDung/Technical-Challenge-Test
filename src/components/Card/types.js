export type Props = {
  link: string,
  imgUrl: string,
  avatarUrl: string,
  username: string,
  toggleModal: Function,
};
