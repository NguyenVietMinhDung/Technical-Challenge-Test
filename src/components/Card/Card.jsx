import React from 'react';
import CSSModules from 'react-css-modules';
import type { Props } from './types';
import styles from './Card.scss';

const Card = (props: Props) => {
  const {
    link, imgUrl, avatarUrl, username, toggleModal,
  } = props;
  const handleClickEvent = imgUrl => () => toggleModal(imgUrl);
  return (
    <div styleName="container">
      <div styleName="card-body">
        <img
          styleName="image"
          src={imgUrl}
          alt=""
          onClick={handleClickEvent(imgUrl)}
        />
        <div styleName="metadata">
          <a styleName="link" href={link}>
            <i className="lni-paperclip size-sm" />
          </a>
        </div>
      </div>
      <div styleName="card-footer">
        {avatarUrl && <img styleName="avatar" src={avatarUrl} alt="" />}
        {username && <span styleName="user">{username}</span>}
      </div>
    </div>
  );
};

export default CSSModules(Card, styles);
