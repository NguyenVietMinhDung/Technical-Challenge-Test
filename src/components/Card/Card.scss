@import '../../styles/variables';

%card-padding {
  padding: 10px;
  box-sizing: border-box;
}

.container {
  padding: 20px;
  display: flex;
  flex-flow: column wrap;
  box-sizing: border-box;

  .card {
    &-body {
      display: flex;
      flex-flow: column wrap;
      background-color: $white-color;

      .image {
        cursor: pointer;
        width: 100%;
        height: 250px;
        background: url($loading-url) 50% no-repeat;

        @media (max-width: $tablet-breakpoint) {
          height: 200px;
        }

        @media (max-width: $mobile-breakpoint) {
          height: 150px;
        }
      }

      .metadata {
        display: flex;
        flex-flow: row wrap;
        margin-top: 10px;

        .link {
          color: $shadow-color;
        }
      }

      @extend %card-padding;
    }

    &-footer {
      width: 100%;
      display: flex;
      flex-flow: row wrap;
      align-items: center;
      background-color: $light-grey-color;

      .avatar {
        width: 25px;
        height: 25px;
        border-radius: 100%;
        margin-right: 5px;
      }

      .user {
        flex: 1;
        white-space: nowrap;
        overflow: hidden;
        text-overflow: ellipsis;
        color: $violet-color;
      }

      @extend %card-padding;
    }
  }
}