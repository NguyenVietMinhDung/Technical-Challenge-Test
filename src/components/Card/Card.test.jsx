import 'jsdom-global/register';
import React from 'react';
import { shallow, mount } from 'enzyme';
import gifImgUrl from '../../assets/giphy.gif';
import Card from './Card';

const defaultProps = {
  link: 'https://developers.giphy.com/docs',
  imgUrl: gifImgUrl,
  avatarUrl: gifImgUrl,
  username: 'Dung VM Nguyen',
  toggleModal: jest.fn(),
};

describe('<Modal />', () => {
  test('renders correctly', () => {
    const component = shallow(<Card {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
  test('handle click event', () => {
    const component = mount(<Card {...defaultProps} />);
    component.find('.card-body > img').simulate('click');
    expect(component.props().toggleModal).toBeCalledWith(component.props().imgUrl);
  });
});
