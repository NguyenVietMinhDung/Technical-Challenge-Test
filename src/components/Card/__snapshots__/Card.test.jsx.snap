// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="card-body"
  >
    <img
      alt=""
      className="image"
      onClick={[Function]}
      src="test-file-stub"
    />
    <div
      className="metadata"
    >
      <a
        className="link"
        href="https://developers.giphy.com/docs"
        target="_blank"
      >
        <i
          className="lni-paperclip size-sm"
        />
      </a>
    </div>
  </div>
  <div
    className="card-footer"
  >
    <img
      alt=""
      className="avatar"
      src="test-file-stub"
    />
    <span
      className="user"
    >
      Dung VM Nguyen
    </span>
  </div>
</div>
`;
